crates/control/src/ball_filter.rs

```diff
@@ -59,26 +62,25 @@ pub struct CycleContext {
 #[context]
 #[derive(Default)]
 pub struct MainOutputs {
-    pub ball_position: MainOutput<Option<BallPosition<Ground>>>,
+    pub ball_position: MainOutput<Option<FilteredBall<Ground>>>,
     pub removed_ball_positions: MainOutput<Vec<Point2<Ground>>>,
     pub hypothetical_ball_positions: MainOutput<Vec<HypotheticalBallPosition<Ground>>>,
 }
 
 impl BallFilter {
     pub fn new(_context: CreationContext) -> Result<Self> {
         Ok(Self {
-            hypotheses: Vec::new(),
+            ball_filter: Default::default(),
         })
     }
 
-    fn persistent_balls_in_control_cycle<'a>(
-        context: &'a CycleContext,
+    fn time_ordered_balls<'a>(
+        balls_top: &'a BTreeMap<SystemTime, Vec<Option<&'a Vec<Ball>>>>,
+        balls_bottom: &'a BTreeMap<SystemTime, Vec<Option<&'a Vec<Ball>>>>,
```

Comment:
huh? how is this related? I cannot follow, can you please elaborate?

Reply:
When I take owned BTreeMaps, I can no longer pass the context by reference to the advance_hypotheses function since it is partially moved. Instead I have to borrow the required subfields separately

Reply:
uff... true... I'm fine with either way. Decide what you think fits best in this situation. I was just annoyed by all the lifetimes :roll_eyes: